src/lib.rs

```diff
@@ -72,12 +74,46 @@ pub trait Database: plumbing::DatabaseOps {
         #![allow(unused_variables)]
     }
 
+    /// Starts unwinding the stack if the current revision is cancelled.
+    ///
+    /// This method can be called by query implementations that perform
+    /// potentially expensive computations, in order to speed up propagation of
+    /// cancellation.
+    ///
+    /// Cancellation will automatically be triggered by salsa on any query
+    /// invocation.
+    ///
+    /// This method should not be overridden by `Database` implementors.
```

Comment:
We should mention that it invokes `salsa_event` with `WillCheckCancellation`, so implementors can override that to intercept.